packages/frontend/src/features/sqlRunner/store/barChartSlice.ts

```diff
@@ -53,6 +56,23 @@ export const barChartConfigSlice = createSlice({
                 config.fieldConfig.x.reference = action.payload;
             }
         },
+        setGroupByReference: (
+            { config },
+            action: PayloadAction<{
+                reference: string;
+            }>,
+        ) => {
+            if (config?.fieldConfig) {
+                config.fieldConfig.groupBy = [
+                    { reference: action.payload.reference },
+                ];
+            }
+        },
+        unsetGroupByReference: ({ config }) => {
```

Comment:
It seems like you did this very deliberately, but I'll ask: why have the unset specifically instead of passing undefined to setGroupByReference?

Reply:
To me it was more explicit but I don't feel strongly

Reply:
I guess I don't either 😄. I was wondering if there was some gotcha, but if its jsut for readability, I'm cool with that.